lib/src/tool_definition.dart

```diff
@@ -212,11 +212,11 @@ class _Snapshot {
   void _snapshotCompleted() => _snapshotCompleter.complete();
 }
 
-/// A singleton that keeps track of cached snapshot files. The [dispose]
+/// A class that keeps track of cached snapshot files. The [dispose]
 /// function must be called before process exit to clean up snapshots in the
 /// cache.
 class SnapshotCache {
-  static SnapshotCache _instance;
+  static final _instances = <ResourceProvider, SnapshotCache>{};
```

Comment:
Gnerally exactly one key/value pair in any given execution, yeah? Maybe like... two, for a test execution that uses mocks and real test packages.

Reply:
Yes.  The cache pretended before that there was only ever one, but I was suspicious of how it handled mocks or other inputs by ignoring them after the first call to createInstance if you hadn't dispose()d of them.  That seemed like a bad idea in an asynchronous program, and was a pain to migrate without a refactor.